src/mysql_conn.erl

```diff
@@ -616,13 +618,18 @@ named_prepare(Name, Query, State) ->
 
 %% @doc Transforms result sets into a structure appropriate to be returned
 %% to the client.
-handle_query_call_result([], _Query, State, []) ->
+handle_query_call_result([_] = Recs, Query, State) ->
+    handle_query_call_result(Recs, single_result, 1, Query, State, []);
```

Comment:
Awesome! Just an idea: Can these two can be combined into one: If it's an integer it means it's a multi-result and for single-results we can use an atom, such as for example `ResultNum :: pos_integer() | not_applicable`. They only have a meaning together anyway and some combinations are impossible (`single_result` with N > 1).

I realize we need an extra case somewhere for the recursion, but on the other hand there will be one parameter less in other places... So you can decide if it's worth it or not.

Reply:
Ok, did as you suggested, no biggie :) I had that in mind as well, but the requirement for an extra `case` was why I ended up using an extra parameter instead. So, is that what you would like it to be like?

Reply:
Yes, I slightly prefer the combined parameter. Nice!